### ISSUE TITLE
Bad STATUS reported for Pod

### ISSUE THREAD

**Issue by caseydavenport** [kubernetes/kubernetes#44298]
**Kubernetes version** (use `kubectl version`):

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.1", GitCommit:"b0b7a323cc5a4a2019b2e9520c21c7830b7f708e", GitTreeState:"clean", BuildDate:"2017-04-03T20:44:38Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.1", GitCommit:"b0b7a323cc5a4a2019b2e9520c21c7830b7f708e", GitTreeState:"clean", BuildDate:"2017-04-03T20:33:27Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: GCE
- **OS** (e.g. from /etc/os-release): debian
- **Install tools**: kubeadm

**What happened**:

`kubectl get pods` resulted in the following status printed to the CLI

```
kube-system   kube-dns-3913472980-j99wd                         0/3       rpc error: code = 2 desc = failed to start container "828b8a9306dfae6cf6a47fdad62a6165adf1f2ae4eb2366772230b8817d0f2e2": Error response from daemon: {"message":"cannot join network of a non running container: 96785a6e586bf7010d3a586a1025e33faee5954ff20c8f4ade7a076ad2a0947f"}   0          32m
```

**What you expected to happen**:

The STATUS field should be "ContainerCreating" or "Error" (or something similarly short)

**How to reproduce it** (as minimally and precisely as possible):

Don't have a reliable repro yet.

**Comment by feborges** [comment]
I'm using kubeadm and it's with the same error.

kubectl get pods --all-namespaces

`kube-system    kube-dns-3913472980-l9lcv       0/3       rpc error: code = 2 desc = failed to start container "6ca41e9b4fdec3736aec177c092cc2219172415dfe0d820baa678838b65d8a36": Error response from daemon: cannot join network of a non running container: 3b691287affd705931dbda62c610fd794ba770e908c0c74738271ff8b185f05e`

**Comment by xiangpengzhao** [comment]
In this case, the `STATUS` is from [container.State.Waiting.Reason](https://github.com/kubernetes/kubernetes/blob/abd92fab1e8c1b638178d58efab7a09509a053c7/pkg/printers/internalversion/printers.go#L301). The Reason is from [here](https://github.com/kubernetes/kubernetes/blob/c51efa9ba0929a643544078d5c182ba75e4b4087/pkg/kubelet/kubelet_pods.go#L1363). One reason might be the content mentioned in this issue. Other reasons are `ErrImagePull`, etc.

@adohe Any optimization suggestions?

**Comment by gtirloni** [comment]
Confirmed on 1.6.4 (CentOS 7.3, Docker 1.12.6).

**Comment by caseydavenport** [comment]
Labeling as SIG-cli as this appears to be a problem with the output generated by kubectl.

/kind bug
/sig cli

**Comment by lucianenache** [comment]
I have the same problem under 1.6.5 and debian jessie with docker 17.5.0

**Comment by ijumps** [comment]
@caseydavenport Hi, I got the same error. It's looks more like a bug for my situation. I create a new issue here: https://github.com/projectcalico/cni-plugin/issues/352

**Comment by pgnaleen** [comment]
root@microServEnv3:~# kubectl get pods --all-namespaces --show-all
NAMESPACE     NAME                                    READY     STATUS                                                                                                                                                                                                                                                                                RESTARTS   AGE
default       busybox                                 0/1       ContainerCannotRun                                                                                                                                                                                                                                                                    0          4h
default       java8-1202446462-kjc5l                  0/1       rpc error: code = 2 desc = failed to start container "e7d1aff4a98d2f4a41cb6907e52fa92dfaa3afa76fda445b4331e11f92ce8124": Error response from daemon: {"message":"cannot join network of a non running container: 7411b4b89f7ac244ffbf07dd4b81d52b5bb9d30b770ac03b0ad294bd52efa40d"}   0          24m
default       mavenname-401845560-qt51w               0/1       rpc error: code = 2 desc = failed to start container "8a59cf6dc4b48e4cd154e5c89b62de09002997da14aa9b55a5251858b1657a41": Error response from daemon: {"message":"cannot join network of a non running container: 4829ed84498dd23fb4858985197947395acae72a3a490b5cb49aa8bf06aec068"}   2          51m
kube-system   etcd-microservenv3                      1/1       Running                                                                                                                                                                                                                                                                               4          1d
kube-system   kube-apiserver-microservenv3            1/1       Running                                                                                                                                                                                                                                                                               13         1d
kube-system   kube-controller-manager-microservenv3   1/1       Running                                                                                                                                                                                                                                                                               10         1d
kube-system   kube-dns-2425271678-25bcq               0/3       rpc error: code = 2 desc = failed to start container "24bf2e60339eee39820ff60659308365322129a74dab2e350983f99bfb83743f": Error response from daemon: {"message":"cannot join network of a non running container: e5fe6853068ea662615195f070aac52aa4ed38556dcc6528aa7f0f033db05361"}   135        1d
kube-system   kube-flannel-ds-slr92                   1/2       CrashLoopBackOff                                                                                                                                                                                                                                                                      120        1d
kube-system   kube-proxy-7bx78                        1/1       Running                                                                                                                                                                                                                                                                               5          1d
kube-system   kube-scheduler-microservenv3            1/1       Running                                                                                                                                                                                                                                                                               8          1d
kube-system   weave-net-rzp8w                         1/2       CrashLoopBackOff                                                                                                                                                                                                                                                                      125        1d
root@microServEnv3:~#

**Comment by pgnaleen** [comment]
![image](https://user-images.githubusercontent.com/6938631/28164578-afe3e070-67ed-11e7-9fbb-6caaeb1c8de7.png)
i cant run a pod they are going to fail. I am new to kubernetes. i think problem may be cni network i can't setup flenal net work or wave net. dns server also not going to show as running. please help me.

**Comment by spnzig** [comment]
@pgnaleen I am facing a similar issue. Have you figured any workaround for this?

**Comment by pgnaleen** [comment]
yes problem was i haven't created funnel network using correct .yml files.

**Comment by pgnaleen** [comment]
https://github.com/coreos/flannel/blob/master/Documentation/Kubernetes.md

use following command if you creat first time the flannel cn if you already setup it use last command.
kubectl apply -f kube-flannel-rbac.yml -f kube-flannel.yml


kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel-rbac.yml

**Comment by rayburgemeestre** [comment]
I had the same issue on a 1.6.4 cluster that I had to fix the certificates for, since they all expired after one year. After that it was a big mess. In my case one of the few things I had overlooked was the tokens, so I basically deleted all secrets from kubectl get secrets -n kube-system
And then all the pods in kube-system. In my case calico would then come up properly again and the pods that were still running but had networking issues started having a functioning network again. 
Also kube-dns and kube-dashboard started working after that finally (they displayed this cannot join error).

I think the tokens may have been generated with the old certificates, and even if you delete them they get auto-generated again but they are not mounted in the pods again (is my guess) so that's why deleting the pods may help
